### PR TITLE
Hide inactive groups on leader removal while adding a confirmation box

### PR DESCRIPTION
**Pull request by HacksForHugs**
Fix: #146

### CODE REVIEW

File: app/assets/javascripts/groups_index.js [hunk 1]
```diff
@@ -37,6 +40,26 @@ if (!top.location.pathname.includes("users")) {
...
+    $button.removeClass("remove-member");
+    $button.text("Join Group");
+    $.ajax({
+      type: "PUT",
+      url: "/groups/" + $button.attr("value") + "/leave",
+      success: function(data) {
+        $(".group_id" + group_id).removeClass("active");
+        $(".group_id" + group_id).removeClass("inactive");
+        $(".group_id" + group_id).addClass("" + data.state);
+        console.log(data.state);
```
Comment (hidesminimally) [inline app/assets/javascripts/groups_index.js]: remove this

File: app/assets/javascripts/groups_index.js [hunk 2]
```diff
@@ -49,16 +72,14 @@ if (!top.location.pathname.includes("users")) {
...
-      $button.text("Join Group");
-
-      $.ajax({
-        type: "PUT",
-        url: "/groups/" + $button.attr("value") + "/leave"
-      });
-
+    if ($button.hasClass("is-member") && $button.hasClass("true")) {
+      var confirmLeave = confirm("You are a leader of this group, if you remove your membership this group will become inactive. Are you sure you want to remove your membership?");
+      console.log(confirmLeave);
```
Comment (hidesminimally) [inline app/assets/javascripts/groups_index.js]: remove this

File: app/controllers/groups_controller.rb [hunk 3]
```diff
@@ -63,7 +63,14 @@ def join
   def leave
     @group = Group.friendly.find(params[:id])
     @group.remove_user(current_user)
-    redirect_to groups_path, notice: "You have successfully left the group"
+    hash = {:state => 'active'}
```
Comment (hidesminimally) [inline app/controllers/groups_controller.rb]: use the newer syntax: `hash = {state: 'active}`

File: app/controllers/groups_controller.rb [hunk 4]
```diff
@@ -63,7 +63,14 @@ def join
...
+    if @group.inactive?
```
Comment (hidesminimally) [inline app/controllers/groups_controller.rb:67]: same here

File: app/controllers/groups_controller.rb (hunk 3 above)
Comment (lilshim) [inline app/controllers/groups_controller.rb]: should change variable name to something more descriptive. perhaps `state`

File: app/assets/javascripts/groups_index.js [hunk 5]
```diff
@@ -37,6 +40,26 @@ if (!top.location.pathname.includes("users")) {
...
 
+  function leave_group(group_id) {
+    $button.removeClass("is-member");
+    $button.removeClass("remove-member");
+    $button.text("Join Group");
+    $.ajax({
+      type: "PUT",
+      url: "/groups/" + $button.attr("value") + "/leave",
+      success: function(data) {
+        $(".group_id" + group_id).removeClass("active");
```
Comment (hidesminimally) [inline app/assets/javascripts/groups_index.js]: don't repeat yourself in your code --   `$(".group_id" + group_id)` can be set to a variable. `$group_div = $(".group_id" + group_id)`

File: app/assets/javascripts/groups_index.js [hunk 6]
```diff
@@ -37,6 +40,26 @@ if (!top.location.pathname.includes("users")) {
...
+        $(".group_id" + group_id).removeClass("inactive");
+        $(".group_id" + group_id).addClass("" + data.state);
```
Comment (hidesminimally) [inline app/assets/javascripts/groups_index.js]: you can condense these three lines to `$div.removeClass("active inactive").addClass(data.state)`

**Pull request merged**